core/src/main/java/hudson/Util.java

```diff
@@ -1168,6 +1168,47 @@ public static FileSet createFileSet(@Nonnull File baseDir, @Nonnull String inclu
         return createFileSet(baseDir,includes,null);
     }
 
+    /**
+     * Creates a symlink to targetPath at baseDir+symlinkPath.
+     *
+     * @param pathForSymlink
+     *      The absolute path of the symlink itself as a path object.
+     * @param fileForSymlink
+     *      The absolute path of the symlink itself as a file object.
+     * @param target
+     *      The path that the symlink should point to. Usually relative to the directory of the symlink but may instead be an absolute path.
+     * @param symlinkPath
+     *      Where to create a symlink in (relative to {@code baseDir})
+     *
+     * Returns true on success
+     */
+    private static boolean createSymlinkAtomic(@Nonnull Path pathForSymlink, @Nonnull File fileForSymlink, @Nonnull Path target, @Nonnull String symlinkPath) {
+        try {
+            File symlink = File.createTempFile("symtmp", null, fileForSymlink);
+            symlink.delete();
+            Path tempSymlinkPath = symlink.toPath();
+            Files.createSymbolicLink(tempSymlinkPath, target);
+            try {
+                Files.move(tempSymlinkPath, pathForSymlink, java.nio.file.StandardCopyOption.ATOMIC_MOVE);
+                return true;
+            } catch (
+                UnsupportedOperationException |
+                SecurityException |
+                IOException e) {
+                // If we couldn't perform an atomic move or the setup, we fall through to another approach
+            }
+            // If we didn't return after our atomic move, then we want to clean up our symlink
+            symlink.delete();
```

Comment:
It uses old File API which actually uses boolean values to return the result. At=nd the result is not checked
https://docs.oracle.com/javase/7/docs/api/java/io/File.html#delete()